### PR TITLE
DCGEO-151: Update pipeline code to use RDA API 0.4 MVP

### PR DESCRIPTION
**Pull request by brianburton**
### Change Details

The RDA API 0.4 MVP added serveral significant changes:

- New repeating payer field in FissClaim
- FissBeneZPayer and FissInsuredPayer messages
- Several new fields were added to FissPayer
- Input parameter changed to require a starting sequence number
- Result objects changed to specific change objects for FISS and MCS claims
- Multi-way (rather than 2 way) enums

This PR updates the RDA API pipeline and mock server to support these new capabilties.  It also adds a DB schema migration file to add a new table for FISS payers and several new columns for FISS claims.

**Note: 20 of the changed files in this PR are proto files from the RDA API release.  These files do not need to be reviewed since they are published by the RDA API team with each release of the RDA API.**

### Acceptance Validation

- Are all RDA API FISS claim and payer changes included?
- Are tests adequate.

### Feedback Requested

Nothing in particular.

### External References

- [DCGEO-151](https://jira.cms.gov/browse/DCGEO-151)
- [RDA API Release Documentation](https://confluenceent.cms.gov/pages/viewpage.action?pageId=194863183)

### Security Implications

No changes affect security they just add new incoming data.

- [ ] new software dependencies

- [ ] altered security controls

No

- [ ] new data stored or transmitted

New FISS claim fields plus new FISS payer records.

- [ ] security checklist is completed for this change

No

- [ ] requires more information or team discussion to evaluate security implications

No

### CODE REVIEW

**Review by KyRobbins — APPROVED** [review]

**Review by KyRobbins — APPROVED** [review]

File: apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/java/gov/cms/bfd/pipeline/rda/grpc/source/FissClaimTransformer.java [hunk 1]
```diff
@@ -223,7 +416,58 @@ private PreAdjFissClaim transformClaim(FissClaim from, DataTransformer transform
...
+            true,
+            1,
+            1,
+            servTypeCd.getEnumString(from),
+            value -> {
+              to.setServTypeCdMapping(PreAdjFissClaim.ServTypeCdMapping.Normal);
+              to.setServTypeCd(value);
+            })
+        .copyEnumAsString(
+            PreAdjFissClaim.Fields.servTypeCd,
```
Comment (ericbuckley) [inline apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/java/gov/cms/bfd/pipeline/rda/grpc/source/FissClaimTransformer.java:433]: are all of these (lines 422-461) supposed to be `PreAdjFissClaim.Fields.servTypeCd`?

**Review by ericbuckley — APPROVED** [review]

File: apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/proto/rda/VERSION.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+0.4.0
```
[the remark this answers is not in the document]
Comment (brianburton) [reply apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/proto/rda/VERSION.txt:1]: I have added a note in the PR description about the files coming from the RDA API team and thus not needing to be reviewed.

I don't think we'll ever pull the proto files from anywhere at build time.  The interfaces there are important to the overall design of our entities and our client code so changes need to at least be reviewed and tied to our code.

File: apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V33__preadj_rda_0_4_table_updates.sql [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+ * new columns to FissClaims for RDA API 0.4.
+ * https://confluenceent.cms.gov/display/MPSM/MVP+0.4+Data+Dictionary+for+Replicated+Data+Access+API
+ *
+ * Where possible column names map directly to fields in the RDA API result message.
+ */
+
+/*
+ * Add new FISS claim fields from RDA API MVP 0.2 release.
+ */
+alter table "pre_adj"."FissClaims"
```
[the remark this answers is not in the document]
Comment (brianburton) [reply apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V33__preadj_rda_0_4_table_updates.sql:12]: Yes

File: apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/java/gov/cms/bfd/pipeline/rda/grpc/RdaChange.java [hunk 4]
```diff
@@ -12,17 +12,21 @@
  */
 @Getter
 public class RdaChange<T> {
+  public static final long MIN_SEQUENCE_NUM = 0;
```
[the remark this answers is not in the document]
Comment (brianburton) [reply apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/java/gov/cms/bfd/pipeline/rda/grpc/RdaChange.java:15]: No this is just a constant for use in places that need to return a value now.  As part of the replay feature development a mechanism will be added to pull the value from the database or a config when a job starts but that will be in addition to this.

File: apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/java/gov/cms/bfd/pipeline/rda/grpc/source/FissClaimTransformer.java (hunk 1 above)
Comment (brianburton) [reply apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/java/gov/cms/bfd/pipeline/rda/grpc/source/FissClaimTransformer.java:433]: Yes they are all referring to the same field by design.  Each of them is one possible outcome of the 4-way oneof field serv_typ_cd.  You'll notice that each of them pulls from a different field in the GRPC FissClaim object.

````protobuf
  // Bill Classification
  oneof serv_typ_cd {
    FissBillClassification serv_type_cd_enum = 44;
    FissBillClassificationForClinics serv_type_cd_for_clinics_enum = 45;
    FissBillClassificationForSpecialFacilities serv_type_cd_for_special_facilities_enum = 46;
    string serv_typ_cd_unrecognized = 47;
  };
````

File: apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/java/gov/cms/bfd/pipeline/rda/grpc/server/RandomFissClaimGenerator.java [hunk 5]
```diff
@@ -1,13 +1,26 @@
 package gov.cms.bfd.pipeline.rda.grpc.server;
 
 import com.google.common.annotations.VisibleForTesting;
+import gov.cms.mpsm.rda.v1.fiss.FissAssignmentOfBenefitsIndicator;
```
Comment (mackec1) [inline apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/java/gov/cms/bfd/pipeline/rda/grpc/server/RandomFissClaimGenerator.java:4]: More curious than anything r.e. the 'v1' in package(s). While good practice to version things regardless of future possibilities, are we expecting that things could change this much?

File: apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/test/java/gov/cms/bfd/pipeline/rda/grpc/server/RandomFissClaimSourceTest.java [hunk 6]
```diff
@@ -34,11 +34,11 @@ public void threeMaxToReturn() throws Exception {
 
     assertEquals(true, source.hasNext());
     claim = source.next();
-    assertEquals("12793", claim.getDcn());
+    assertTrue(claim.getDcn().length() > 0);
```
Comment (mackec1) [inline apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/test/java/gov/cms/bfd/pipeline/rda/grpc/server/RandomFissClaimSourceTest.java:37]: assuming this change reflects the 'randomness'  of the claims genearation.

**Review by mackec1 — APPROVED** [review]
Lot to digest here although it is mostly boiler-plate changes. Things look reasonable although it is difficult to really dig through everything based on the size of the PR.

File: apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/java/gov/cms/bfd/pipeline/rda/grpc/server/RandomFissClaimGenerator.java (hunk 5 above)
Comment (brianburton) [reply apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/java/gov/cms/bfd/pipeline/rda/grpc/server/RandomFissClaimGenerator.java:4]: The `FissAssignmentOfBenefitsIndicator` enum is generated by the protobuf compiler from the proto files provided by the RDA team.  So the package naming for them corresponds to the naming within their API.

File: apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/test/java/gov/cms/bfd/pipeline/rda/grpc/server/RandomFissClaimSourceTest.java (hunk 6 above)
Comment (brianburton) [reply apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/test/java/gov/cms/bfd/pipeline/rda/grpc/server/RandomFissClaimSourceTest.java:37]: Correct.  The big thing being tested here is just that exactly three claims were returned.  Having specific value checks for Dcn there just made the tests brittle without adding any benefit.

**Pull request merged**